### PR TITLE
refactor(compiler): remove old ngtools api and add listLazyRoutes to …

### PR DESCRIPTION
**Pull request by tbosch**
…new api

Usages of `NgTools_InternalApi_NG_2` from `@angular/compiler-cli` will now
throw an error.

Adds `listLazyRoutes` to `@angular/compiler-cli/ngtools2.ts` for getting
the lazy routes of a `ng.Program`.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: packages/compiler-cli/src/main.ts [hunk 1]
```diff
@@ -28,15 +28,15 @@ export function main(
   let {project, rootNames, options, errors: configErrors, watch, emitFlags} =
       config || readNgcCommandLineAndConfiguration(args);
   if (configErrors.length) {
-    return reportErrorsAndExit(configErrors, consoleError);
+    return reportErrorsAndExit(configErrors, undefined, consoleError);
```
Comment (chuckjaz) [inline packages/compiler-cli/src/main.ts]: Put a `/* options */` in front of `undefined` to indicate what parameter you are skipping.

**Review by chuckjaz — APPROVED** [review]

File: packages/compiler-cli/src/main.ts (hunk 1 above)
Comment (tbosch) [reply packages/compiler-cli/src/main.ts]: Done

**Comment by tbosch** [comment]
Fails now on the expected error `Error: Please update @angular/cli. Angular 5+ requires at least Angular CLI 1.5+`.

**Review by hansl — APPROVED** [review]

**Comment by tbosch** [comment]
Regarding the approval for root: I talked to @IgorMinar about them, so this PR is good to go.

**Comment by tbosch** [comment]
This is also green in G3.

**Review by mhevery — APPROVED** [review]

File: aio/package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
   "license": "MIT",
   "scripts": {
     "aio-use-local": "node tools/ng-packages-installer overwrite . --debug --ignore-packages @angular/service-worker",
-    "aio-use-npm": "node tools/ng-packages-installer restore .",
+    "aio-use-npm": "node tools/ng-packages-installer restore . && yarn upgrade @angular/cli@1.3.0",
```
Comment (gkalpak) [inline aio/package.json]: What is this change for? Is this supposed to use the same version as currently specified in the lockfile?
Can we remove the changes in `aio/package.json` once we upgrade to `@angular/cli@1.5.0-rc.2` (or higher)?

**Comment by IgorMinar** [comment]
once we upgrade to cli rc.3 we can remove this bit.

**Pull request closed without merging**